packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1153,6 +1179,17 @@ export class TransactionController extends BaseController<
         },
       );
 
+      addedTransactionMeta = updateBridgeTypesTransaction(
```

Comment:
When we added this support for traditional swaps, it was for backwards compatibility as we were migrating the extension and mobile controllers together.

But if all this ultimately does is add some swap or bridge specific properties to the metadata, then could we remove all this coupling and extra logic and just call `updateTransaction` from the client and populate those fields explicitly once `addTransaction` has resolved and added the new metadata?

If we want more type safety, could we just add a simple `updateBridgeTransaction` method that can only update a given list of properties but it's the responsibility of the client to call it?

Reply:
Made it so the client will call `updateTransaction` and attach the desired fields.